src/openrct2/scripting/ScTile.hpp

```diff
@@ -0,0 +1,1112 @@
+            dukglue_register_property(ctx, &ScTileElement::edgeStyle_get, &ScTileElement::edgeStyle_set, "edgeStyle");
+            dukglue_register_property(ctx, &ScTileElement::grassLength_get, &ScTileElement::grassLength_set, "grassLength");
+            dukglue_register_property(ctx, &ScTileElement::hasOwnership_get, nullptr, "hasOwnership");
+            dukglue_register_property(ctx, &ScTileElement::hasConstructionRights_get, nullptr, "hasConstructionRights");
+            dukglue_register_property(ctx, &ScTileElement::ownership_get, &ScTileElement::ownership_set, "ownership");
+            dukglue_register_property(ctx, &ScTileElement::parkFences_get, &ScTileElement::parkFences_set, "parkFences");
+
+            // Track only
+            dukglue_register_property(ctx, &ScTileElement::trackType_get, &ScTileElement::trackType_set, "trackType");
+            dukglue_register_property(ctx, &ScTileElement::hasChainLift_get, &ScTileElement::hasChainLift_set, "isChainLift");
+
+            // Small Scenery only
+            dukglue_register_property(ctx, &ScTileElement::age_get, &ScTileElement::age_set, "age");
+
+            // Footpath only
+            dukglue_register_property(ctx, &ScTileElement::railings_get, &ScTileElement::railings_set, "railings");
+            dukglue_register_property(
+                ctx, &ScTileElement::edgesAndCorners_get, &ScTileElement::edgesAndCorners_set, "edgesAndCorners");
+            dukglue_register_property(
+                ctx, &ScTileElement::slopeDirection_get, &ScTileElement::slopeDirection_set, "slopeDirection");
+            dukglue_register_property(ctx, &ScTileElement::isQueue_get, &ScTileElement::isQueue_set, "isQueue");
+            dukglue_register_property(
+                ctx, &ScTileElement::queueBannerDirection_get, &ScTileElement::queueBannerDirection_set,
+                "queueBannerDirection");
+
+            dukglue_register_property(
+                ctx, &ScTileElement::isBlockedByVehicle_get, &ScTileElement::isBlockedByVehicle_set, "isBlockedByVehicle");
+            dukglue_register_property(ctx, &ScTileElement::isWide_get, &ScTileElement::isWide_set, "isWide");
+
+            dukglue_register_property(ctx, &ScTileElement::addition_get, &ScTileElement::addition_set, "addition");
+            dukglue_register_property(
+                ctx, &ScTileElement::additionStatus_get, &ScTileElement::additionStatus_set, "additionStatus");
+            dukglue_register_property(
+                ctx, &ScTileElement::isAdditionBroken_get, &ScTileElement::isAdditionBroken_set, "isAdditionBroken");
+            dukglue_register_property(
+                ctx, &ScTileElement::isAdditionGhost_get, &ScTileElement::isAdditionGhost_set, "isAdditionGhost");
+
+            // Entrance only
+            dukglue_register_property(
+                ctx, &ScTileElement::footpathObject_get, &ScTileElement::footpathObject_set, "footpathObject");
+        }
+    };
+
+    class ScTile
```

Comment:
Class is using snake_cast instead of CamelCase.

Should be `ScriptTile` (file as well)

Reply:
My paradigm for the case here was that it matched the casing of the JavaScript API followed then by a get or set suffix. An exceptional case if you will for these bindings between the scripting language and the game source code.

The `Sc` prefix was also trying to distinguish between an interface represented in the JavaScript API as opposed to any class in OpenRCT2. For example `ScriptEngine` is not a binding in the scripting API but you might think it was if all these other classes were renamed to `Script*`.

Reply:
I see, it's a little sad to have it deviating from what we propose as coding guidelines, but if you think they're actually that confuse, I can accept it.

One way to differentiate would be to use the namespaces and not have `using ...` namespaces for all of them everywhere, but this will give you a lot of work in this huge patch.

Reply:
I am not sure. It was a deliberate decision to make it as obvious as possible that this represents a JavaScript plugin script object binding. This is why the functions are lowerCamal and snake case (since properties aren't in C++), to keep it as close to the plugin form as possible.

The packet handling functions in Network.cpp also use snake case presumably to help show the close relationship to the packet enum type.

Reply:
Hmm I see, if there's reasoning behind it, I'm ok with that, I just wanted to make sure it wasn't an automagic naming.